azurerm/internal/services/machinelearning/machine_learning_compute_cluster_resource_test.go

```diff
@@ -0,0 +1,206 @@
+		{
+			Config: r.basic(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("identity.#").HasValue("1"),
+				check.That(data.ResourceName).Key("identity.0.type").HasValue("SystemAssigned"),
+				check.That(data.ResourceName).Key("identity.0.principal_id").Exists(),
+				check.That(data.ResourceName).Key("identity.0.tenant_id").Exists(),
+				check.That(data.ResourceName).Key("scale_settings.#").HasValue("1"),
+				check.That(data.ResourceName).Key("scale_settings.0.max_node_count").Exists(),
+				check.That(data.ResourceName).Key("scale_settings.0.min_node_count").Exists(),
+				check.That(data.ResourceName).Key("scale_settings.0.node_idle_time_before_scale_down").Exists(),
+			),
+		},
+		data.RequiresImportErrorStep(r.requiresImport),
+	})
+}
+
+func (r ComputeClusterResource) Exists(ctx context.Context, client *clients.Client, state *terraform.InstanceState) (*bool, error) {
+	computeClusterClient := client.MachineLearning.MachineLearningComputeClient
+	id, err := parse.ComputeClusterID(state.ID)
+
+	if err != nil {
+		return nil, err
+	}
+
+	computeResource, err := computeClusterClient.Get(ctx, id.ResourceGroup, id.WorkspaceName, id.ComputeName)
+	if err != nil {
+		if utils.ResponseWasNotFound(computeResource.Response) {
+			return utils.Bool(false), nil
+		}
+		return nil, fmt.Errorf("retrieving Machine Learning Compute Cluster %q: %+v", state.ID, err)
+	}
+	return utils.Bool(computeResource.Properties != nil), nil
+}
+
+func (r ComputeClusterResource) basic(data acceptance.TestData) string {
+	template := r.template(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_machine_learning_compute_cluster" "test" {
+  name                          = "CC-%d"
+  location                      = azurerm_resource_group.test.location
+  vm_priority                   = "LowPriority"
+  vm_size                       = "STANDARD_DS2_V2"
+  machine_learning_workspace_id = azurerm_machine_learning_workspace.test.id
+  subnet_resource_id            = azurerm_subnet.test.id
+
+  scale_settings {
+	min_node_count = 0
+	max_node_count = 1
+	node_idle_time_before_scale_down = "PT30S" # 30 seconds
+  }
+
+  identity {
+    type = "SystemAssigned"
+  }
+}
+`, template, data.RandomIntOfLength(8))
+}
+
+func (r ComputeClusterResource) requiresImport(data acceptance.TestData) string {
+	template := r.basic(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_machine_learning_compute_cluster" "import" {
+	name                          = azurerm_machine_learning_compute_cluster.test.name
+	location                      = azurerm_machine_learning_compute_cluster.test.location
+	vm_priority                   = azurerm_machine_learning_compute_cluster.test.vm_priority
+	vm_size                       = azurerm_machine_learning_compute_cluster.test.vm_size
+	machine_learning_workspace_id = azurerm_machine_learning_compute_cluster.test.id
```

Comment:
This or similar should fix it probably 
```suggestion
	machine_learning_workspace_id = azurerm_machine_learning_workspace.test.id
```

Reply:
Thanks, fixed now :)